src/wallet/mod.rs

```diff
@@ -2176,6 +2183,40 @@ pub(crate) mod test {
         assert_eq!(drain_output.value, 30_000 - details.fee.unwrap_or(0));
     }
 
+    #[test]
+    fn test_create_tx_drain_to_and_utxos() {
+        let (wallet, _, _) = get_funded_wallet(get_test_wpkh());
+        let addr = wallet.get_address(New).unwrap();
+        let utxos: Vec<_> = wallet
+            .get_available_utxos()
+            .unwrap()
+            .into_iter()
+            .map(|(u, _)| u.outpoint)
+            .collect();
+        let mut builder = wallet.build_tx();
+        builder
+            .drain_to(addr.script_pubkey())
+            .add_utxos(&utxos)
```

Comment:
Not very pressing but maybe we can have another test, but instead of preselected utxos, set `drain_wallet` and `drain_to` and confirm behavior, to verify the other side of the conditional?

Reply:
Do you mean like `test_create_tx_drain_wallet_and_drain_to` (already in master: https://github.com/bitcoindevkit/bdk/pull/625/files/098c448dfbdf1922d9911760da27d76df67af420#diff-18f5545e0b6d6f7d1d8ef9f744c9d6d1458baaffc419e69f826626c00dddc260L2138), or something else?